lib/interfaces/src/ADCInterface.cpp

```diff
@@ -0,0 +1,130 @@
+  adc0_offsets.at(_adc_parameters.channels.rl_suspot_channel)         = _adc_parameters.offsets.rl_suspot_offset;
+  adc0_offsets.at(_adc_parameters.channels.rr_suspot_channel)         = _adc_parameters.offsets.rr_suspot_offset;
+
+  adc1_scales.at(_adc_parameters.channels.thermistor0_channel)        = _adc_parameters.scales.thermistor0_scale;
+  adc1_scales.at(_adc_parameters.channels.thermistor1_channel)        = _adc_parameters.scales.thermistor1_scale;
+  adc1_scales.at(_adc_parameters.channels.thermistor2_channel)        = _adc_parameters.scales.thermistor2_scale;
+  adc1_scales.at(_adc_parameters.channels.thermistor3_channel)        = _adc_parameters.scales.thermistor3_scale;
+  adc1_scales.at(_adc_parameters.channels.thermistor4_channel)        = _adc_parameters.scales.thermistor4_scale;
+  adc1_scales.at(_adc_parameters.channels.thermistor5_channel)        = _adc_parameters.scales.thermistor5_scale;
+  adc1_scales.at(_adc_parameters.channels.thermistor6_channel)        = _adc_parameters.scales.thermistor6_scale;
+  adc1_scales.at(_adc_parameters.channels.thermistor7_channel)        = _adc_parameters.scales.thermistor7_scale;
+
+  adc1_offsets.at(_adc_parameters.channels.thermistor0_channel)        = _adc_parameters.offsets.thermistor0_offset;
+  adc1_offsets.at(_adc_parameters.channels.thermistor0_channel)        = _adc_parameters.offsets.thermistor1_offset;
+  adc1_offsets.at(_adc_parameters.channels.thermistor0_channel)        = _adc_parameters.offsets.thermistor2_offset;
+  adc1_offsets.at(_adc_parameters.channels.thermistor0_channel)        = _adc_parameters.offsets.thermistor3_offset;
+  adc1_offsets.at(_adc_parameters.channels.thermistor0_channel)        = _adc_parameters.offsets.thermistor4_offset;
+  adc1_offsets.at(_adc_parameters.channels.thermistor0_channel)        = _adc_parameters.offsets.thermistor5_offset;
+  adc1_offsets.at(_adc_parameters.channels.thermistor0_channel)        = _adc_parameters.offsets.thermistor6_offset;
+  adc1_offsets.at(_adc_parameters.channels.thermistor0_channel)        = _adc_parameters.offsets.thermistor7_offset;
+
+  _adc0.emplace(
+    _adc_parameters.pinouts.adc0_spi_cs_pin, 
+    MCP_ADC_DEFAULT_SPI_SDI, 
+    MCP_ADC_DEFAULT_SPI_SDO, 
+    MCP_ADC_DEFAULT_SPI_CLK,
+    MCP_ADC_DEFAULT_SPI_SPEED,
+    adc0_scales.data(),
+    adc0_offsets.data()
+  );
+
+  _adc1.emplace(
+    _adc_parameters.pinouts.adc1_spi_cs_pin, 
+    MCP_ADC_DEFAULT_SPI_SDI, 
+    MCP_ADC_DEFAULT_SPI_SDO, 
+    MCP_ADC_DEFAULT_SPI_CLK,
+    MCP_ADC_DEFAULT_SPI_SPEED,
+    adc1_scales.data(),
+    adc1_offsets.data()
+  );
+}
+
+void ADCInterface::tick_adc0() { _adc0->tick(); }
+void ADCInterface::tick_adc1() { _adc1->tick(); }
+
+const ADCInterfaceParams_s& ADCInterface::get_adc_params() const {
+    return _adc_parameters;
+}
+
+AnalogConversion_s ADCInterface::read_glv() {
+  return _adc0->data.conversions.at(_adc_parameters.channels.glv_sense_channel);
```

Comment:
iirc if you're using -> with optionals, the behavior can be undefined and possible cause seg faults. i would use .value()?